model.py

```diff
@@ -3256,7 +3256,9 @@ def all_editions(self, recursion_level=5):
     def all_identifier_ids(self, recursion_level=5):
         _db = Session.object_session(self)
         primary_identifier_ids = [
-            x.primary_identifier.id for x in self.editions]
+            x.primary_identifier.id for x in self.editions
+            if x.primary_identifier
+        ]
```

Comment:
If a Work is associated with an Edition that has no primary identifier, that's weird, but that in itself shouldn't prevent `all_identifier_ids` from working.